src/bio/bio_wal.c

```diff
@@ -1777,6 +1788,24 @@ bio_wal_replay(struct bio_meta_context *mc,
 		rc = unmap_wal(mc, unmap_start, unmap_end, NULL);
 		if (rc)
 			D_ERROR("Unmap after replay failed. "DF_RC"\n", DP_RC(rc));
+
+		/* create and set replay metrics */
+		clock_gettime(CLOCK_REALTIME, &tms);
+		e_us = (tms.tv_sec * 1000000) + (tms.tv_nsec / 1000);
+		uuid_unparse_lower(mc->mc_wal->bic_pool_id, pool);
+		ret = d_tm_add_metric(&wal_sz, D_TM_GAUGE, "WAL replay size", "bytes",
+				      "pool/%s/tgt_%d/wal_replay/size", pool, trg);
+		if (ret)
+			D_WARN("Failed to create WAL size replay telemetry: "DF_RC"\n", DP_RC(ret));
+		ret = d_tm_add_metric(&wal_et, D_TM_GAUGE, "WAL replay time", "us",
```

Comment:
use D_TM_DURATION? see vam_epr_dur for refrences.

Reply:
Plz see the update. Thanks!

Reply:
@wangshilong I changed D_TM_DURATION back to GAUGE and clock_gettime() bcz of d_tm_mark_duration_start/end seem not be safe with ULT context switch in bio_wal_replay/load_wal when per-engine metrics is obtained. Please advise if opposite is true.